src/rpc/rawtransaction.cpp

```diff
@@ -850,15 +850,20 @@ UniValue SignTransaction(CMutableTransaction& mtx, const UniValue& prevTxsUnival
     // transaction to avoid rehashing.
     const CTransaction txConst(mtx);
     // Sign what we can:
+    CAmount inout_amount = 0;
+    bool known_inputs = true;
     for (unsigned int i = 0; i < mtx.vin.size(); i++) {
         CTxIn& txin = mtx.vin[i];
         const Coin& coin = view.AccessCoin(txin.prevout);
         if (coin.IsSpent()) {
+            known_inputs = false;
```

Comment:
You should also set `known_input = false` when `sigdata.witness` below is false (unless it's known that every input is spending a witness output, there is no guarantee that providing incorrect amounts will result in invalid signatures).

Reply:
I'm not sure what you mean by "providing incorrect amounts". This is directly from the UTXO set of the node itself. When would it tell itself incorrect amounts?

Reply:
`view` is not the node's UTXO set, but a local variable which holds a few cached UTXO entries. Those *can* come from the UTXO set or mempool, but could also have been provided by the user.

Reply:
Makes sense, thanks. I updated the code to only work with segwit inputs (i.e. added the change you requested).